Some Functions in Camera

AdornedJoker opened the pull request:
This has almost all the functions except for,
 **setImageServerViewNoLerp**,  **characterZoom**,  **getNewZoomDistance**

AdornedJoker committed:
some work on camera

AdornedJoker committed:
headers

AdornedJoker committed:
moved some functions to header

AdornedJoker committed:
alright

AdornedJoker committed:
Update Camera.h

AdornedJoker committed:
first time with doing reflections

bluepilledgreat commented on Client/App/v8datamodel/Camera.cpp:
```diff
@@ -0,0 +1,213 @@
+		{
+			cameraExternallyAdjusted = true;
+			
+			#if defined(_DEBUG) || defined(_RELEASEASSERT)
```
This is lazy. Figure out how to do this assertion properly.
```
if (Math::legalCameraCoord(cameraGoal))
{
	gCamera.setCoordinateFrame(cameraGoal);
}
else
{
	RBXASSERT(0);
}
```

This is the equivalent code of that.

bluepilledgreat commented:
> That's alright! Should I close this PR then?

No, keep it open. Once the camera files are committed into the master branch, you can fix any merge conflicts that arise from it.
So far, this PR looks good.

bluepilledgreat commented on Client/App/include/v8datamodel/Camera.h:
```diff
@@ -0,0 +1,121 @@
+		static float distanceMinOcclude();
+		static float getNewZoomDistance(float, float);
+
+		static const Reflection::EnumPropDescriptor<Camera, CameraType> desc_cameraType;
```
Why are you adding these to the header file? In the future, please don't. Resym does not show any of these static fields existing. When doing reflection, you don't need to add stuff to the class's definition.

AdornedJoker committed:
the removal of descriptors in the header

AdornedJoker committed:
changed assertion to be less lazy

AdornedJoker committed:
styling!

AdornedJoker committed:
moved vars into if block

less ugly

AdornedJoker committed:
lowercased and made code less ugly

AdornedJoker committed:
removed something unnecessary

AdornedJoker committed:
Camera::Camera()

AdornedJoker committed:
Camera::setCameraSubject

AdornedJoker committed:
more styling

AdornedJoker committed:
forgot some

AdornedJoker committed:
merge conflict fix attempt

I moved G3D. I was having compiler errors when having it in between Instance and Extents.

AdornedJoker committed:
Merge branch 'master' into somethingCamera

AdornedJoker committed:
Merge branch 'RBLXDecomp:master' into somethingCamera

AdornedJoker committed:
More Functions!

AdornedJoker committed:
agonizing work

AdornedJoker committed:
i gave up

ThePlayerRolo commented:
> I'm going to be honest, this is the best I can do with this. I'm already at my limit.
> 
> I'm just unable to do all of it.

i think you should be able to merge without everything. Progress is still progress

AdornedJoker commented:
I hope so!

AdornedJoker committed:
Merge branch 'master' into somethingCamera

AdornedJoker committed:
Update App.vcproj

AdornedJoker committed:
ok

AdornedJoker committed:
zoomExtents

87.82% matching

AdornedJoker committed:
updateGoal matching to 87.80%

AdornedJoker committed:
tryZoomExtents and headers

AdornedJoker committed:
simplification

AdornedJoker committed:
setDistanceFromTarget

AdornedJoker committed:
Merge branch 'master' into somethingCamera

AdornedJoker committed:
Update App.vcproj
